modules/common/math/linear_interpolation.cc

```diff
@@ -91,22 +98,18 @@ TrajectoryPoint InterpolateUsingLinearApproximation(const TrajectoryPoint &tp0,
   double t1 = tp1.relative_time();
 
   TrajectoryPoint tp;
-  tp.set_v(common::math::lerp(tp0.v(), t0, tp1.v(), t1, t));
-  tp.set_a(common::math::lerp(tp0.a(), t0, tp1.a(), t1, t));
+  tp.set_v(lerp(tp0.v(), t0, tp1.v(), t1, t));
+  tp.set_a(lerp(tp0.a(), t0, tp1.a(), t1, t));
   tp.set_relative_time(t);
 
   PathPoint *path_point = tp.mutable_path_point();
-  path_point->set_x(common::math::lerp(pp0.x(), t0, pp1.x(), t1, t));
-  path_point->set_y(common::math::lerp(pp0.y(), t0, pp1.y(), t1, t));
-  path_point->set_theta(
-      common::math::lerp(pp0.theta(), t0, pp1.theta(), t1, t));
-  path_point->set_kappa(
-      common::math::lerp(pp0.kappa(), t0, pp1.kappa(), t1, t));
-  path_point->set_dkappa(
-      common::math::lerp(pp0.dkappa(), t0, pp1.dkappa(), t1, t));
-  path_point->set_ddkappa(
-      common::math::lerp(pp0.ddkappa(), t0, pp1.ddkappa(), t1, t));
-  path_point->set_s(common::math::lerp(pp0.s(), t0, pp1.s(), t1, t));
+  path_point->set_x(lerp(pp0.x(), t0, pp1.x(), t1, t));
+  path_point->set_y(lerp(pp0.y(), t0, pp1.y(), t1, t));
+  path_point->set_theta(lerp(pp0.theta(), t0, pp1.theta(), t1, t));
```

Comment:
Shall we use slerp to set theta? because theta is an angle.

Reply:
Yes, the original code seems to have a bug here.